File: client/app/components/ReactFittyHelper.ts
import * as ReactDOM from "react-dom";
import fitty from "fitty";

export function fitHeaderNode(headerNode, maxFontHeight, minFontHeight) {
    headerNode.childNodes.forEach(node => {
        fitty(node, {
            maxSize: maxFontHeight,
            minSize: minFontHeight,
            multiLine: true
        })
    })
}

export function fitHeaderTextOnCardComponent(maxFontHeight, minFontHeight, component, className: any) {
    const node = ReactDOM.findDOMNode(component);
    let headerNode = node.getElementsByClassName(className)[0];

    fitHeaderNode(headerNode, maxFontHeight, minFontHeight);
}

File: client/src/main/kotlin/com/zegreatrob/coupling/client/ReactFittyHelper.kt
package com.zegreatrob.coupling.client

import org.w3c.dom.Node
import kotlin.js.json

@JsModule("fitty")
@JsNonModule
private external val fitty: dynamic

fun Node.fitHeaderNode(maxFontHeight: Double, minFontHeight: Double) {
    fitty.default(this, json(
            "maxSize" to maxFontHeight,
            "minSize" to minFontHeight,
            "multiLine" to true
    )).unsafeCast<Unit>()
}